internal/services/autopi_api_service.go

```diff
@@ -160,6 +162,30 @@ func (a *autoPiAPIService) UnassociateDeviceTemplate(deviceID string, templateID
 	return nil
 }
 
+// GetVehicleLoggers Get slice of loggers from vehicleID, populates parameter "loggers *[]AutoPiVehicleLogger"
+func (a *autoPiAPIService) GetVehicleLoggers(vehicle AutoPiDongleVehicle, loggers *[]AutoPiVehicleLogger) error {
```

Comment:
Change the input parameter to be just the autopiVehicleID. No need to pass in the entire object since in the code all you're using is the `vehicle.ID` field. 
Also, change the ` loggers *[]AutoPiVehicleLogger` from being an input parameter to being returned, eg. `(*[]AutoPiVehicleLogger, error), and then on line 176 just return the loggers.